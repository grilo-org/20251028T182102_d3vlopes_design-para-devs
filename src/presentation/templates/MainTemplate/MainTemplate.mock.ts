import { footerMock } from '@/presentation/components/molecules/Footer/Footer.mock';
import { heroMock } from '@/presentation/components/molecules/Hero/Hero.mock';

import { authorSectionMock } from './sections/Author/AuthorSection.mock';
import { bonusSectionMock } from './sections/Bonus/BonusSection.mock';
import { chaptersSectionMock } from './sections/Chapters/ChaptersSection.mock';
import { checkoutSectionMock } from './sections/Checkout/CheckoutSection.mock';
import { faqSectionMock } from './sections/Faq/FaqSection.mock';
import { introSectionMock } from './sections/Intro/IntroSection.mock';
import { previewSectionMock } from './sections/Preview/PreviewSection.mock';
import { projectSectionMock } from './sections/Project/ProjectSection.mock';
import { rememberCheckoutSectionMock } from './sections/RememberCheckout/RememberCheckoutSection.mock';

import { MainTemplateProps } from '.';

export const mainTemplateMock: MainTemplateProps = {
	hero: heroMock,
	intro: introSectionMock,
	chapters: chaptersSectionMock,
	author: authorSectionMock,
	preview: previewSectionMock,
	checkout: checkoutSectionMock,
	project: projectSectionMock,
	bonus: bonusSectionMock,
	rememberCheckout: rememberCheckoutSectionMock,
	faq: faqSectionMock,
	footer: footerMock,
};
