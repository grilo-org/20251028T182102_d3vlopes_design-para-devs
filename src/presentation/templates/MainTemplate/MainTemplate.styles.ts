import { styleWrapper } from '@/libs/styles/tailwind-variants';

const styles = styleWrapper({
	slots: {
		containerStyles: 'max-w-[78.4287rem] px-4 xl:px-0',
		fullPageStyles: 'w-full',
		wrapperStyles: 'flex flex-col gap-20 mx-auto md:gap-[10.1875rem]',
		authorSectionStyles: 'bg-background-400 my-20 md:mt-[13rem]',
		previewSectionStyles: 'max-w-[93.75rem] mb-20 md:mb-[13rem]',
		checkoutSectionStyles:
			'mb-20 md:mb-[10.1875rem] lg:px-[5rem] lg:max-w-[78.4287rem]',
		bonusSectionStyles: 'max-w-[93.75rem] my-20 md:my-[10.1875rem]',
		footerStyles: 'mt-20 md:mt-[10.1875rem]',
	},
});

const makeStyles = styles();

export const classNames = {
	containerStyles: makeStyles.containerStyles(),
	fullPageStyles: makeStyles.fullPageStyles(),
	wrapperStyles: makeStyles.wrapperStyles(),
	authorSectionStyles: makeStyles.authorSectionStyles(),
	previewSectionStyles: makeStyles.previewSectionStyles(),
	checkoutSectionStyles: makeStyles.checkoutSectionStyles(),
	bonusSectionStyles: makeStyles.bonusSectionStyles(),
	footerStyles: makeStyles.footerStyles(),
};
