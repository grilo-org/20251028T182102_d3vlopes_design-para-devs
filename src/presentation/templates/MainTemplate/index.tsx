import { MainModel } from '@/domain/models/Main';

import { cn } from '@/utils/cn';

import { Footer } from '@/presentation/components/molecules/Footer';
import { Hero } from '@/presentation/components/molecules/Hero';

import * as Section from './helpers/exports/sections';

import { classNames } from './MainTemplate.styles';

export type MainTemplateProps = MainModel;

export const MainTemplate = (props: MainTemplateProps) => {
	return (
		<main>
			<Hero {...props.hero} />

			<div className={cn(classNames.wrapperStyles, classNames.containerStyles)}>
				<Section.IntroSection {...props.intro} />
				<Section.ChaptersSection {...props.chapters} />
			</div>

			<div
				className={cn(
					classNames.fullPageStyles,
					classNames.authorSectionStyles,
				)}
			>
				<Section.AuthorSection {...props.author} />
			</div>

			<div
				className={cn(
					classNames.wrapperStyles,
					classNames.previewSectionStyles,
				)}
			>
				<Section.PreviewSection {...props.preview} />
			</div>

			<div
				className={cn(
					classNames.wrapperStyles,
					classNames.checkoutSectionStyles,
				)}
			>
				<Section.CheckoutSection {...props.checkout} />
			</div>

			<div className={cn(classNames.wrapperStyles, classNames.containerStyles)}>
				<Section.ProjectSection {...props.project} />
			</div>

			<div
				className={cn(classNames.wrapperStyles, classNames.bonusSectionStyles)}
			>
				<Section.BonusSection {...props.bonus} />
			</div>

			<div className={cn(classNames.wrapperStyles, classNames.containerStyles)}>
				<Section.RememberCheckoutSection {...props.rememberCheckout} />
				<Section.FaqSection {...props.faq} />
			</div>

			<Footer className={classNames.footerStyles} {...props.footer} />
		</main>
	);
};
