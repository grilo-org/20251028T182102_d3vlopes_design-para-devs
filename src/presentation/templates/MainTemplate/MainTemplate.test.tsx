import { render, screen } from '@testing-library/react';

import './helpers/test/mock';

import { MockDataTestID } from './helpers/test/mockDataTestID';

import { mainTemplateMock as mock } from './MainTemplate.mock';

import { MainTemplate } from '.';

const { getByTestId } = screen;

describe('<MainTemplate />', () => {
	beforeEach(() => render(<MainTemplate {...mock} />));

	it('should render Hero', () => {
		const hero = getByTestId(MockDataTestID.HERO);

		expect(hero).toBeInTheDocument();
	});

	it('should render intro section', () => {
		const introSection = getByTestId(MockDataTestID.INTRO_SECTION);

		expect(introSection).toBeInTheDocument();
	});

	it('should render chapters section', () => {
		const chaptersSection = getByTestId(MockDataTestID.CHAPTERS_SECTION);

		expect(chaptersSection).toBeInTheDocument();
	});

	it('should render author section', () => {
		const authorSection = getByTestId(MockDataTestID.AUTHOR_SECTION);

		expect(authorSection).toBeInTheDocument();
	});

	it('should render preview section', () => {
		const previewSection = getByTestId(MockDataTestID.PREVIEW_SECTION);

		expect(previewSection).toBeInTheDocument();
	});

	it('should render checkout section', () => {
		const checkoutSection = getByTestId(MockDataTestID.CHECKOUT_SECTION);

		expect(checkoutSection).toBeInTheDocument();
	});

	it('should render project section', () => {
		const projectSection = getByTestId(MockDataTestID.PROJECT_SECTION);

		expect(projectSection).toBeInTheDocument();
	});

	it('should render bonus section', () => {
		const bonusSection = getByTestId(MockDataTestID.BONUS_SECTION);

		expect(bonusSection).toBeInTheDocument();
	});

	it('should render remember checkout section', () => {
		const rememberCheckoutSection = getByTestId(
			MockDataTestID.REMEMBER_CHECKOUT_SECTION,
		);

		expect(rememberCheckoutSection).toBeInTheDocument();
	});

	it('should render faq section', () => {
		const faqSection = getByTestId(MockDataTestID.FAQ_SECTION);

		expect(faqSection).toBeInTheDocument();
	});

	it('should render Footer', () => {
		const footer = getByTestId(MockDataTestID.FOOTER);

		expect(footer).toBeInTheDocument();
	});

	it('should match snapshot', () => {
		const { container } = render(<MainTemplate {...mock} />);

		expect(container).toMatchSnapshot();
	});
});
