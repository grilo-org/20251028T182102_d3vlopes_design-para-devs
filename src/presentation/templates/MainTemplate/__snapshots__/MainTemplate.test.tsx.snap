// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainTemplate /> should match snapshot 1`] = `
<div>
  <main>
    <div
      data-testid="mock-hero"
    />
    <div
      class="flex flex-col gap-20 mx-auto md:gap-[10.1875rem] max-w-[78.4287rem] px-4 xl:px-0"
    >
      <div
        data-testid="mock-intro-section"
      />
      <div
        data-testid="mock-chapters-section"
      />
    </div>
    <div
      class="w-full bg-background-400 my-20 md:mt-[13rem]"
    >
      <div
        data-testid="mock-author-section"
      />
    </div>
    <div
      class="flex flex-col gap-20 mx-auto md:gap-[10.1875rem] max-w-[93.75rem] mb-20 md:mb-[13rem]"
    >
      <div
        data-testid="mock-preview-section"
      />
    </div>
    <div
      class="flex flex-col gap-20 mx-auto md:gap-[10.1875rem] mb-20 md:mb-[10.1875rem] lg:px-[5rem] lg:max-w-[78.4287rem]"
    >
      <div
        data-testid="mock-checkout-section"
      />
    </div>
    <div
      class="flex flex-col gap-20 mx-auto md:gap-[10.1875rem] max-w-[78.4287rem] px-4 xl:px-0"
    >
      <div
        data-testid="mock-project-section"
      />
    </div>
    <div
      class="flex flex-col gap-20 mx-auto md:gap-[10.1875rem] max-w-[93.75rem] my-20 md:my-[10.1875rem]"
    >
      <div
        data-testid="mock-bonus-section"
      />
    </div>
    <div
      class="flex flex-col gap-20 mx-auto md:gap-[10.1875rem] max-w-[78.4287rem] px-4 xl:px-0"
    >
      <div
        data-testid="mock-remember-checkout-section"
      />
      <div
        data-testid="mock-faq-section"
      />
    </div>
    <div
      data-testid="mock-footer"
    />
  </main>
</div>
`;
